Bio/SeqIO/QualityIO.py

```diff
@@ -1100,36 +1100,25 @@ def __next__(self) -> SeqRecord:
 
         if not quality_string.isascii():
             # Look for invalid non-ascii characters
-            invalid_char_indices = [
-                i for i, char in enumerate(quality_string) if not char.isascii()
-            ]
-            if not invalid_char_indices:
+            for i, char in enumerate(quality_string):
+                if not char.isascii():
+                    bad_index = i
+                    break
+            else:
                 raise ValueError(
```

Comment:
RuntimeError? Because if this happens, it is a bug in Python.

Reply:
It wouldn't necessarily a bug in Python, though testing it right now, it would probably trip the length check before getting to this point

Example:
```
from Bio.SeqIO.QualityIO import FastqPhredIterator
with open(r"C:\Users\Flaar\PycharmProjects\biopythonTest\data\solexa_non_ascii.fastq", "rt") as handle:
    for i, rec in enumerate(FastqPhredIterator(handle), 1):
        print(rec)
```

Where the file is
```
@SLXA-B3_649_FC8437_R1_1_1_610_79
GATGTGCAATACCTTTGTAGAGGAA
+SLXA-B3_649_FC8437_R1_1_1_610_79
YYYYYYYYYWYYäWWYYYWYWYWW
```
Note: I had to remove an extra quality characters to avoid the length check since ä encodes into two bytes in unicode

Gives me:
```
Traceback (most recent call last):
  File "C:\Users\Flaar\PycharmProjects\biopythonTest\biopython_testing\parsing\ascii_test.py", line 6, in <module>
    for i, rec in enumerate(FastqPhredIterator(handle), 1):
  File "C:\Users\Flaar\miniconda3\envs\biopython\Lib\site-packages\Bio\SeqIO\QualityIO.py", line 1112, in __next__
    raise ValueError(
ValueError: Invalid character (Ã) in quality string is not an ASCII character with context: ...YWYYÃ¤WWY...
```

Reply:
But if this error happens, it probably is both an error with our code and an error with their file, so it also makes some sense to make it a ValueError. I guess RuntimeError could work too, but it's fairly non-specific. Maybe I should really throw two errors here, one from the internal error, which is being caused by the value error from the file